### PR TITLE
Add safari driver

### PR DESCRIPTION
**Pull request by Andmedoctopus**

### CODE REVIEW

**Comment by Andmedoctopus** [comment]
@fsouza @jsfehler could you take a look, please?

**Comment by fsouza** [comment]
> @fsouza @jsfehler could you take a look, please?
> 

Is there any way to write tests for this? Maybe we make it run only on macOS in GHA.

**Comment by Andmedoctopus** [comment]
> > @fsouza @jsfehler could you take a look, please?
> 
> Is there any way to write tests for this? Maybe we make it run only on macOS in GHA.

I added MacOS to CI. Seems Splinter has some bug with Safari. I will fix them and let you know when its done

**Comment by Andmedoctopus** [comment]
@fsouza Please take a look. CI almost stable

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ jobs:
     runs-on: ubuntu-latest
 
     strategy:
+      max-parallel: 3
```
Comment (fsouza) [inline .github/workflows/main.yml:23]: curious on why we're making this change?

File: tests/click_elements.py [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 # Use of this source code is governed by a BSD-style
 # license that can be found in the LICENSE file.
+from .fake_webapp import EXAMPLE_APP
 
 
 class ClickElementsTest:
     def test_click_links(self):
         self.browser.links.find_by_text("FOO").click()
         self.assertIn("BAR!", self.browser.html)
+        self.browser.visit(EXAMPLE_APP)
```
Comment (fsouza) [inline tests/click_elements.py:11]: why did we need to make this change?

File: tests/test_webdriver_remote.py [hunk 3]
```diff
@@ -51,9 +52,169 @@ def setUp(self):
...
+
+    def test_support_with_statement(self):
+        """
+        Remote should support with statement
+        See setup browser
+        """
+
+    def test_should_be_able_to_change_user_agent(self):
+        "Remote should not support custom user agent"
+        pass
```
Comment (fsouza) [inline tests/test_webdriver_remote.py:86]: can we explicitly skip it with a message?

Comment (Andmedoctopus) [reply tests/test_webdriver_remote.py]: Added

File: tests/click_elements.py (hunk 2 above)
Comment (Andmedoctopus) [reply tests/click_elements.py]: The visit in `setUp` was unstable for some reason and page just stuck. Now seems ok. Removed all `visit` in tests

File: .github/workflows/main.yml (hunk 1 above)
Comment (Andmedoctopus) [reply .github/workflows/main.yml]: CI was unstable. Removed all limitation. Works ok

**Comment by Andmedoctopus** [comment]
@fsouza I have fixed all remarks. Could you take a look, please?

**Comment by Andmedoctopus** [comment]
@fsouza could you take a look to the PR, please?

**Review by fsouza — APPROVED** [review]
@Andmedoctopus thanks for the ping, sorry I somehow missed the earlier notification. This looks good, thanks for contributing!

**Comment by Andmedoctopus** [comment]
@fsouza thank you! Could you release a new version of splinter, please?

**Comment by Andmedoctopus** [comment]
> @fsouza thank you! Could you release a new version of splinter, please?

@fsouza, could you make a new release, please?

**Comment by fsouza** [comment]
Will tag a new release once #1229 passes.

**Pull request merged**